src/GraphQL/Types/Scalars/ScalarGraphType.cs

```diff
@@ -50,11 +50,11 @@ public abstract class ScalarGraphType : GraphType
         public virtual object? ParseLiteral(GraphQLValue value) => value switch
         {
             GraphQLBooleanValue b => ParseValue(b.BoolValue.Boxed()),
-            GraphQLIntValue i => ParseValue((string)i.Value), //ISSUE:allocation
-            GraphQLFloatValue f => ParseValue((string)f.Value), //ISSUE:allocation
-            GraphQLStringValue s => ParseValue((string)s.Value), //ISSUE:allocation
+            GraphQLIntValue i => Int.Parse(i.Value),
+            GraphQLFloatValue f => Double.Parse(f.Value),
```

Comment:
We should call the FloatGraphType.ParseLiteral method which has the correct error logic for overflowed numbers.

Reply:
Done. I moved method into `ScalarGraphType`: `ParseDoubleAccordingSpec`